Stylelint: Apply font rules

tbradsha opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

In particular, this undisables and fixes the following:

* `font-weight-notation` ("numeric" is what WP uses): https://stylelint.io/user-guide/rules/font-weight-notation/
* `font-family-name-quotes` ("always-where-recommended" is what WP uses): https://stylelint.io/user-guide/rules/font-family-name-quotes/
* `font-family-no-duplicate-names` (one instance): https://stylelint.io/user-guide/rules/font-family-no-duplicate-names/
* `font-family-no-missing-generic-family-keyword`: https://stylelint.io/user-guide/rules/font-family-no-missing-generic-family-keyword/

For the last one, most instances were icon fonts (e.g. `social-logos`, `dashicons`, `vaultpressregular`, etc.), so I just did a generic family of `sans-serif`. For real fonts, I looked them up and all but one were `sans-serif`, so I used that. I set the two instances of `Recoleta` to `serif`. Oh, and in one CRM instance there was a presumably intentional invalid font-family, but the CSS selector no longer existed, so I just chopped the block.

See also:
* https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/#values

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
*

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

Is the Stylelint CI check still happy?

tbradsha committed:
Undisable font rules

tbradsha committed:
Fix font-weight-notation

tbradsha committed:
Fix font-family-name-quotes

tbradsha committed:
Don't use quotes in font-family when generating CSS file

tbradsha committed:
Fix font-family-no-duplicate-names

tbradsha committed:
Fix font-family-no-missing-generic-family-keyword (most instances)

tbradsha committed:
Fix font-family-no-missing-generic-family-keyword (Recoleta)

tbradsha committed:
Fix font-family-no-missing-generic-family-keyword (vaultpressregular)

tbradsha committed:
Remove unused CSS block

tbradsha committed:
Respect rules when generating CSS

tbradsha committed:
Add changelogs

anomiex commented on projects/packages/classic-theme-helper/src/custom-post-types/css/nova-font.css:
```diff
@@ -1,22 +1,22 @@
 #dashboard_right_now .nova-menu-count a:before,
 #dashboard_right_now .nova-menu-count span:before {
-	font-family: 'nova-font';
+	font-family: nova-font, sans-serif;
```
This one is kind of pointless too. The font contains only two characters, space and U+E603, and seems intended for use only with the rule just below that does `content: '\e603'`.

anomiex commented on projects/plugins/crm/sass/_ZeroBSCRM.prevadmincore.scss:
```diff
@@ -3999,14 +3999,6 @@ table [class*=span],.row-fluid table [class*=span] {
     background-color: #c4e3f3;
 }
 
-#ZeroBSCRMAdminPage .hide-text {
```
?

tbradsha replied:
This is the one I mentioned that no longer matches any selector but probably was intentionally a bad font family. It's been the same way since it was introduced 9 years ago, and `hide-text` is no longer a class used anywhere in the CRM codebases, so I just chopped it altogether.

anomiex commented on projects/plugins/jetpack/modules/carousel/swiper-bundle.css:
We may want to just ignore this copy-pasted file.

tbradsha replied:
c569d18ea3

anomiex commented on projects/plugins/protect/assets/jetpack-protect.css:
```diff
@@ -1,21 +1,21 @@
 [class^="jp-protect-icon"], [class*=" jp-protect-icon"] {
 	/* use !important to prevent issues with browser extensions that change fonts */
-	font-family: 'jetpack-protect' !important;
+	font-family: jetpack-protect, sans-serif !important;
```
Another single-icon font.

anomiex commented on projects/plugins/vaultpress/nav-styles.css:
```diff
@@ -1,18 +1,18 @@
 	content: '\f219';
 	font-size: 18px !important;
-	font-family: 'vaultpressregular' !important;
+	font-family: vaultpressregular, sans-serif !important;
```
Another single-icon font.

anomiex reviewed (commented):
I note that Gutenberg does like
```
/* stylelint-disable-next-line font-family-no-missing-generic-family-keyword -- dashicons don't need a generic family keyword. */
```
We may want to consider doing the same where we use Dashicons, Noticons, Genericons, FontAwesome, social-logos, or any other icon font.

The [ignoreFontFamilies](https://stylelint.io/user-guide/rules/font-family-no-missing-generic-family-keyword/#ignorefontfamilies-regex-regex-string) option might help, I think setting that can let us skip the individual disables for the common ones.

tbradsha committed:
Revert "Fix font-family-no-missing-generic-family-keyword (most instances)"

This reverts commit e9b4714bd6625d66ce7dd1e401c8fad71fbf5137.

tbradsha committed:
Ignore styles in third-party file

tbradsha committed:
Add generic font-family fallback to real fonts

tbradsha committed:
Ignore one-off icon fonts with comment

tbradsha committed:
Remove fallbacks from icon fonts

tbradsha committed:
Dashicons → dashicons to match core

tbradsha committed:
genericons → Genericons per examples in font docs

tbradsha committed:
Ignore lint errors on some specific icon fonts

tbradsha commented:
For icon fonts I've now:
* removed generic font-family fallbacks
* ignored style violations with comments on the one-offs
* added the remainder to the `ignoreFontFamilies` param

tbradsha committed:
Remove now-irrelevant changelog

anomiex commented on projects/plugins/crm/sass/_ZeroBSCRM.prevadmincore.scss:
```diff
@@ -97,7 +97,7 @@ pre {
 }
 
 code,kbd,pre,samp {
-    font-family:monospace,monospace;
```
We may want to keep this.

In some browsers, `font-family: monospace` renders smaller, apparently because those browsers have separate preferences for "default proportional font size" and "default monospace font size", and so if the font-size winds up being based on the browser default (including the `1em` here) then it applies the corresponding preference to calculate it. Apparently doing `font-family: monospace, monospace` tricks at least some such browsers to use the proportional preference instead of the monospace preference.

When I tried it out, Firefox 137 and Chromium 135 both show different rendering for `font-family: monospace` versus `font-family: monospace, monospace`.

tbradsha replied:
Restored and disabled linting for that one instance: a7e2158c25

anomiex reviewed (dismissed):
One last comment. LGTM if you think the change in rendering I pointed out is ok.

tbradsha committed:
Restore rendering hack

tbradsha committed:
Merge branch 'trunk' into fix/stylelint/font_rules
